File: code/frontend/src/features/kanban/service/taskService.js
import axiosInstance from '@api/axiosConfig'

const unwrap = (response) => response.data?.data

export const taskService = {
  getProjectTasks: async (projectId) => {
    const response = await axiosInstance.get(`/v1/projects/${projectId}/tasks`)
    return unwrap(response) || []
  },

  getTask: async (taskId) => {
    const response = await axiosInstance.get(`/v1/tasks/${taskId}`)
    return unwrap(response)
  },

  getMyTasks: async () => {
    const response = await axiosInstance.get('/v1/my-tasks')
    return unwrap(response) || []
  },

  createTask: async (projectId, payload) => {
    const response = await axiosInstance.post(`/v1/projects/${projectId}/tasks`, payload)
    return unwrap(response)
  },

  updateTask: async (taskId, payload) => {
    const response = await axiosInstance.put(`/v1/tasks/${taskId}`, payload)
    return unwrap(response)
  },

  deleteTask: async (taskId) => {
    await axiosInstance.delete(`/v1/tasks/${taskId}`)
  },

  updateTaskStatus: async (taskId, status, blockedReason) => {
    const response = await axiosInstance.patch(`/v1/tasks/${taskId}/status`, { status, blockedReason })
    return unwrap(response)
  },

  updateTaskAssignee: async (taskId, assigneeId) => {
    const response = await axiosInstance.patch(`/v1/tasks/${taskId}/assignee`, { assigneeId })
    return unwrap(response)
  },
}

export default taskService

File: code/backend/src/main/java/org/example/backend/service/TaskService.java
package org.example.backend.service;

import org.example.backend.dto.*;

import java.time.LocalDate;
import java.util.List;

public interface TaskService {
    List<TaskResponse> getProjectTasks(Long projectId, Long userId);
    List<TaskResponse> getMyTasks(Long userId);
    TaskResponse getTask(Long taskId, Long userId);
    TaskResponse createTask(Long projectId, TaskRequest request, Long userId);
    TaskResponse updateTask(Long taskId, TaskRequest request, Long userId);
    TaskResponse updateTaskStatus(Long taskId, TaskStatusUpdateRequest request, Long userId);
    TaskResponse updateTaskAssignee(Long taskId, TaskAssigneeUpdateRequest request, Long userId);
    void deleteTask(Long taskId, Long userId);

    // ── Daily / Weekly View ───────────────────────────────────────────────────
    DailyViewResponse getDailyView(Long projectId, Long userId, LocalDate date);
    WeeklyViewResponse getWeeklyView(Long projectId, Long userId, LocalDate weekStart);
}
